docs/concepts/architecture/master-node-communication.md

```diff
@@ -106,3 +108,83 @@ cluster (connecting to the ssh server listening on port 22) and passes all
+| --kubernetes-service-node-port |                   | If non-zero, the Kubernetes master service (which apiserver creates/maintains) will be of type NodePort, using this as the value of the port. If zero, the Kubernetes master service will be of type ClusterIP. |
+| --secure-port                  | 6443              | The port on which to serve HTTPS with authentication and authorization. Set to 0 to disable |
+| --service-node-port-range      | 30000-32767       | A port range to reserve for services with NodePort visibility. |
+
+
+### Communicating / Authenticating with the API Servers
+All services will use the following arguments to specify how to communicate with the API servers.
+| Argument     | Default                     | Description                              |
+| ------------ | --------------------------- | ---------------------------------------- |
+| --master     |                             | The address of the Kubernetes API server (overrides any value in kubeconfig) |
+| --kubeconfig | /var/lib/kubelet/kubeconfig | Path to a kubeconfig file, specifying how to connect / authenticate to the API server. |
+
+### kube-controler-manager
+
+| Argument  | Default |
+| --------- | ------- |
+| --address | 0.0.0.0 |
+| --port    | 10252   |
+
+### cloud-controler-manager
+
+| Argument  | Default |
+| --------- | ------- |
+| --address | 0.0.0.0 |
+| --port    | 10253   |
+
+
+### kube-scheduler
+
+| Argument  | Default |
+| --------- | ------- |
+| --address |         |
+| --port    | 10251   |
+
+
+### kubelet
+
+> Note that kubelet does not support the `--master` flag
+
+| Argument               | Default                     | Description                              |
+| ---------------------- | --------------------------- | ---------------------------------------- |
+| --address              | 0.0.0.0                     | The IP address for the Kubelet to serve on |
+| --cadvisor-port        | 4194                        | Set to 0 to disable                      |
+| --docker-endpoint      | unix:///var/run/docker.sock |                                          |
+| --healthz-bind-address | 127.0.0.1                   | Set to 0.0.0.0 for all interfaces        |
+| --healthz-port         | 10248                       | Set to 0 to disable                      |
+| --pod-manifest-path    |                             | Path to the directory containing pod manifest files to run, or the path to a single pod manifest file. Files starting with dots will be ignored. |
+| --port                 | 10250                       | The port for the Kubelet to serve on.    |
+| --read-only-port       | 10255                       | The read-only port for the Kubelet to serve on with no authentication/authorization. Set to 0 to disable |
+
+### kube-proxy
```

Comment:
This is one of the first components that is moving to a config file. The flags are still accepted, but we have marked them as deprecated. It would be worthwhile to document the relevant sections of the config file. I started a PR to auto-generate a sample kube-proxy config yaml file with comments for each field, but it hasn't been merged (#52198).

Reply:
Once the PR merges and their a documentation page I can link to, I will add in the note and link